### ISSUE TITLE
Integrate WebCaster Live DJ Option

### ISSUE THREAD

**Issue by paddatrapper** [LibreTime/libretime#1033]
This somewhat ties in with #480. We should look at integrating [WebCaster](https://github.com/webcast/webcaster)/[webcast.js](https://github.com/webcast/webcast.js) into LibreTime. I image this to be a standalone page that presents a DJ with a microphone input, jingles palette and live cart playlist and allows them to enter their DJ username and password to connect to a show mount. There could be advanced options to connect to the master mount too.

It would be great if the playlist could be populated from the music already in the LibreTime library and the jingles palette loaded with existing jingles, station imaging, etc.

**Comment by zklosko** [comment]
I was recently asking about this. Bonus points if y'all can make a desktop client in Electron that ties in with LibreTime's API. :)

**Comment by paddatrapper** [comment]
The limitation I see here (assuming it uses the master/show mounts) is downloading the library and streaming it to the server again. That would require either a very fast internet connection or pre-downloading tracks, which limits the flexibility in what can be played. Ideally this would use #480 and have tracks played on the server and just stream the mic when necessary. But just integrating WebCaster with a DJ sign-in to send to the default show mount settings (with the option for the user to change it if the admin has changed the settings) would be a useful first step.

**Comment by Robbt** [comment]
Yeah I played with the webcaster integrated into Azuracast and it was cool but also annoying. The ultimate in terms of simplicity for the end-user would be a seamless merge of their local tracks and the remote tracks but this would also be the most complicated thing to configure. I'm going to re-open #480, whenever a topic comes up that was marked stale I'm going to make it a point of re-opening it. This I think is a nice compromise between having too many issues we aren't going to get too and reminding us what has still been affecting users at some point in the last 6 months.

As far as Electron goes, I think they tended to be bloated apps but it might make sense, might would be easier than forking and redesigning BUTT, but I don't know. Do you know any of the libraries for integrating sound/recording in electron apps ? I've also been inclined to just do things straight in javascript, but I guess that is essentially what you are doing on electron anyways, but just bundling a new instance of chrome to run it on.

**Comment by paddatrapper** [comment]
You can also label issues as pinned to prevent lock bot from closing them again

**Comment by zklosko** [comment]
@paddatrapper @Robbt I don't see the electron app as being anything more than a controller that would connect to LibreTime and send `play`, `pause`, `skip`, and `stop after next` commands. Maybe it has a clock and the show's playlist too, but it's just sending control commands.
